interpreter/CMakeLists.txt

```diff
@@ -580,3 +580,35 @@ mark_as_advanced(FORCE BUG_REPORT_URL BUILD_CLANG_FORMAT_VS_PLUGIN BUILD_SHARED_
                        C_INCLUDE_DIRS DEFAULT_SYSROOT FFI_INCLUDE_DIR FFI_LIBRARY_DIR
                        GCC_INSTALL_PREFIX LIBCLANG_BUILD_STATIC TOOL_INFO_PLIST)
 mark_as_advanced(CLEAR LLVM_ENABLE_ASSERTIONS LLVM_BUILD_TYPE)
+
+if(NOT LLVM_TABLEGEN_EXE)
+    message(STATUS "llvm-tblgen not found. Attempting to build llvm-tblgen.")
+
+    # Set LLVM source and build directories
+    set(LLVM_SOURCE_DIR "" CACHE PATH "Path to LLVM source directory")
+    set(LLVM_BINARY_DIR "" CACHE PATH "Path to LLVM build directory")
+
+    # Check if LLVM source and build directories are provided
+    if(LLVM_SOURCE_DIR AND LLVM_BINARY_DIR)
+        # Build llvm-tblgen
+        add_custom_target(llvm_tblgen
+            COMMAND ${CMAKE_COMMAND} -G "${CMAKE_GENERATOR}" .
+            WORKING_DIRECTORY ${LLVM_BINARY_DIR}/tools/llvm-tblgen
+            COMMENT "Building llvm-tblgen"
+        )
+        add_custom_command(TARGET llvm_tblgen
+            COMMAND ${CMAKE_COMMAND} --build .
+            WORKING_DIRECTORY ${LLVM_BINARY_DIR}/tools/llvm-tblgen
+            COMMENT "Building llvm-tblgen"
+        )
+        if(MSVC)
+            set(LLVM_TABLEGEN_EXE ${LLVM_BINARY_DIR}/$<CONFIG>/bin/llvm-tblgen.exe CACHE PATH "Path to llvm-tblgen" FORCE)
+        else()
+            set(LLVM_TABLEGEN_EXE ${LLVM_BINARY_DIR}/bin/llvm-tblgen CACHE PATH "Path to llvm-tblgen" FORCE)
+        endif()
+    else()
+        message(FATAL_ERROR "LLVM source and build directories not provided. Please set LLVM_SOURCE_DIR and LLVM_BINARY_DIR.")
+    endif()
+endif()
+
+add_subdirectory(cling/include/cling/Interpreter)
```

Comment:
I assume this will break Cling standalone builds; this `add_subdirectory` has to move into Cling's build system logic...